CrewChiefV4/UserInterface/MainWindow.cs

```diff
@@ -223,14 +232,23 @@ private void FormMain_Load(object sender, EventArgs e)
                                     float.TryParse(doc.Descendants("soundpackversion").First().Value, out latestSoundPackVersion);
                                     float.TryParse(doc.Descendants("drivernamesversion").First().Value, out latestDriverNamesVersion);
                                     float.TryParse(doc.Descendants("personalisationsversion").First().Value, out latestPersonalisationsVersion);
-                                    baseSoundPackDownloadLocation = doc.Descendants("basesoundpackurl").First().Value;
-                                    baseDriverNamesDownloadLocation = doc.Descendants("basedrivernamesurl").First().Value;
-                                    basePersonalisationsDownloadLocation = doc.Descendants("basepersonalisationsurl").First().Value;
-                                    updateSoundPackDownloadLocation = doc.Descendants("updatesoundpackurl").First().Value;
-                                    update2SoundPackDownloadLocation = doc.Descendants("update2soundpackurl").First().Value;
-                                    updateDriverNamesDownloadLocation = doc.Descendants("updatedrivernamesurl").First().Value;
-                                    updatePersonalisationsDownloadLocation = doc.Descendants("updatepersonalisationsurl").First().Value;
-                                    update2PersonalisationsDownloadLocation = doc.Descendants("update2personalisationsurl").First().Value;
+
+
+                                    baseSoundPackDownloadLocation = MainWindow.getDocDescendantValue(doc, "basesoundpackurl");
```

Comment:
we have Utilities class and this function should better go there, don't you think?

Reply:
Agreed